File: archive/CP/ATCoder/abc373/B2.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <string>
#include <vector>

struct data
{
    
};

auto solution()
{
    std::string K = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"; 
    std::string S;
    std::vector<int> D;
    int d = 0;

    std::getline(std::cin, S);

    for (int i = 0; i < K.size(); i++) {
        for (int j = 0; j < S.size(); j++) {
            if (S[j] == K[i]) {
                D.push_back(j);
            }
        }
    }

    /* for (int x : D) {
        std::cout << x << " ";
    }*/

    for (int i = 0; i < D.size() - 1; i++) {
        d += abs(D[i + 1] - D[i]);
    }

    return d;
}

int main()
{
    /* data d = solution(); */
    auto result = solution();

    std::cout << result;
    
    return 0;
}

File: archive/CP/ATCoder/abc373/B2.py
def solution():
    K = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    S = input()
    D = []
    d = 0

    for i in range(len(K)):
        for j in range(len(S)):
            if (S[j] == K[i]):
                D.append(j)
    
    """ for i in range(len(D)):
        print(D[i], end=' ')
    """

    for i in range(len(D) - 1):
        d += abs(D[i+1] - D[i]) 

    return d

def main():
    result = solution()
    print(result)


if __name__ == "__main__":
    main()